CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(sol2)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+include_directories(.)
```

Comment:
I prefer using `target_include_directories`

Reply:
As it is for all target I prefer a global one.
You talk me about https://cmake.org/cmake/help/latest/variable/CMAKE_INCLUDE_CURRENT_DIR.html I will try to dig it.

Reply:
Even though its for all, there's an argument surrounding the idea of using the single header to build, too. If we have the single header it's a good test to use it to verify it works the same as the full tree.

Reply:
But the single one can only be generated if python3 is found.

Reply:
@ThePhD expressed in gitter he would like to see building implemented with the single header, too. Thus, if it's generated, we should be checking that they build with both the single and full tree. 

If this is running in CI, ie Travis or appveyor, we should make configuration fail if it can't find python.